modules/apps/27-interchain-accounts/controller/client/cli/tx.go

```diff
@@ -101,18 +112,39 @@ appropriate relative timeoutTimestamp must be provided with flag {relative-packe
 				}
 			}
 
-			relativeTimeoutTimestamp, err := cmd.Flags().GetUint64(flagRelativePacketTimeout)
+			timeoutTimestamp, err := cmd.Flags().GetUint64(flagPacketTimeoutTimestamp)
+			if err != nil {
+				return err
+			}
+
+			absoluteTimeouts, err := cmd.Flags().GetBool(flagAbsoluteTimeouts)
 			if err != nil {
 				return err
 			}
 
-			msg := types.NewMsgSendTx(owner, connectionID, relativeTimeoutTimestamp, icaMsgData)
+			// NOTE: relative timeouts using block height are not supported.
+			if !absoluteTimeouts {
+				if timeoutTimestamp == 0 {
+					return errors.New("relative timeouts must provide a non zero value timestamp")
+				}
+
+				// use local clock time as reference time for calculating timeout timestamp.
+				now := time.Now().UnixNano()
+				if now <= 0 {
+					return errors.New("local clock time is not greater than Jan 1st, 1970 12:00 AM")
+				}
+
+				timeoutTimestamp = uint64(now) + timeoutTimestamp
+			}
+
+			msg := types.NewMsgSendTx(owner, connectionID, timeoutTimestamp, icaMsgData)
```

Comment:
Isn't `NewMsgSendTx` expects a relative timeout here, but we passed an absolute one?

Reply:
Ah yea, I think you are correct @yihuang. Good catch. This cli should be revert to use relative timestamps only and included in a patch release.

Did this cause some problem for you?